Added functionality to Md. Data Converter to allow some data transformation

DeepInTheCode opened the pull request:
As part of the 2014 Houston Hackathon, the team I was on found your tool to be an ideal candidate for our use, with some added functionality.  We added the ability to exclude, reorder, and rename columns.

DeepInTheCode committed:
Update index.html

Added static mapping fields

DeepInTheCode committed:
Update index.html

Fixing padding on input elements

DeepInTheCode committed:
Update index.html

Added manual transform checkbox

DeepInTheCode committed:
Update README.md

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update converter.js

Changed DataConverter.prototype.convert function to return headerNames

DeepInTheCode committed:
Update converter.js

Changed return headerNames to attribute of object

dshefman committed:
filtering out fields

dshefman committed:
Merge branch 'gh-pages' of https://github.com/DeepInTheCode/mr-data-converter into gh-pages

dshefman committed:
adding comments

DeepInTheCode committed:
Update Controller.js

Returns headernames from convert function

dshefman committed:
not bi-directional mapping

dshefman committed:
Merge branch 'gh-pages' of https://github.com/DeepInTheCode/mr-data-converter into gh-pages

DeepInTheCode committed:
Update Controller.js

headerNames variable created for document.ready

DeepInTheCode committed:
Update converter.js

Added applyMap function

DeepInTheCode committed:
Update applyMap.js

changed function name to CamelCase

DeepInTheCode committed:
Update mapping-tests.js

Changed function to camelCase

DeepInTheCode committed:
Update converter.js

checking targets

DeepInTheCode committed:
Update converter.js

checking intermediateText

DeepInTheCode committed:
Update converter.js

checking transformedText with JSON.stringify

DeepInTheCode committed:
Update converter.js

create array of strings in targets

dshefman committed:
no targets

dshefman committed:
Merge branch 'gh-pages' of https://github.com/DeepInTheCode/mr-data-converter into gh-pages

dshefman committed:
CamelCase

DeepInTheCode committed:
Update converter.js

checking targets

dshefman committed:
Update applyMap.js

adding logging

dshefman committed:
Update applyMap.js

Log targets

DeepInTheCode committed:
Update Controller.js

headerNames is broken

dshefman committed:
Update applyMap.js

more logging

dshefman committed:
Update applyMap.js

logging input

DeepInTheCode committed:
Update converter.js

intermediateText

DeepInTheCode committed:
Update converter.js

JSON.parse

dshefman committed:
Update applyMap.js

removing logs

DeepInTheCode committed:
Update converter.js

hardcode JSON as output

DeepInTheCode committed:
Create jsonToCsv.js

DeepInTheCode committed:
Update index.html

Add csvToJson.js

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update index.html

corrected filename - jsonToCsv.js

DeepInTheCode committed:
Update converter.js

Added all other outputDataTypes

DeepInTheCode committed:
Update converter.js

Fixed method name ConvertToCSV

DeepInTheCode committed:
Update jsonToCsv.js

Adding headers

DeepInTheCode committed:
Update jsonToCsv.js

removed header functionality

DeepInTheCode committed:
Update jsonToCsv.js

adding headers

DeepInTheCode committed:
Update jsonToCsv.js

added GetHeaders function

DeepInTheCode committed:
Update jsonToCsv.js

Fixed GetHeaders function

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Create addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update converter.js

Create text inputs

DeepInTheCode committed:
Update index.html

add js file for text inputs

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

Fixed widths of boxes

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update addTextInputs.js

add Options

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update index.html

dshefman committed:
fixes to targets and mapping

dshefman committed:
Merge branch 'gh-pages' of https://github.com/DeepInTheCode/mr-data-converter into gh-pages

Conflicts:
	js/applyMap.js

DeepInTheCode committed:
Update index.html

added all authors to meta tag

DeepInTheCode committed:
Update converter.css

Changed color scheme

DeepInTheCode committed:
Update converter.css

DeepInTheCode committed:
Update converter.css

DeepInTheCode committed:
Update converter.css

DeepInTheCode committed:
Update converter.css

DeepInTheCode committed:
Update converter.css

DeepInTheCode committed:
Update Controller.js

Adding targets at page load

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

removing debug alert box

DeepInTheCode committed:
Update converter.js

removing debugging alert box

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update applyMap.js

commented out lines before hash map

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update DataGridRenderer.js

subtract 1 from row num to exclude added headers

DeepInTheCode committed:
Update addTextInputs.js

escaping html tags

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update applyMap.js

Adding debugging

DeepInTheCode committed:
Update applyMap.js

stringify debugging

DeepInTheCode committed:
Update jsonToCsv.js

adding debugging

DeepInTheCode committed:
Update jsonToCsv.js

adding debugging

DeepInTheCode committed:
adding debugging

DeepInTheCode committed:
fixing line

DeepInTheCode committed:
Update jsonToCsv.js

getting only first line

DeepInTheCode committed:
Update applyMap.js

turning off debugging

DeepInTheCode committed:
Update DataGridRenderer.js

fixed rows

DeepInTheCode committed:
Update DataGridRenderer.js

fixed rows on AS

DeepInTheCode committed:
Update DataGridRenderer.js

adding debugging AS

DeepInTheCode committed:
Update DataGridRenderer.js

fixed debugging expression

DeepInTheCode committed:
Update DataGridRenderer.js

DeepInTheCode committed:
Update converter.js

add debugging

DeepInTheCode committed:
Update converter.js

add debugging

DeepInTheCode committed:
Update jsonToCsv.js

test

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

DeepInTheCode committed:
Update jsonToCsv.js

removing test

DeepInTheCode committed:
Update converter.js

turning off debugging

DeepInTheCode committed:
Update applyMap.js

allow no targets

DeepInTheCode committed:
Update applyMap.js

fix variable name targets

DeepInTheCode committed:
Update Controller.js

allow no targets

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update Controller.js

DeepInTheCode committed:
Update applyMap.js

DeepInTheCode committed:
Update applyMap.js

remove no targets code

DeepInTheCode committed:
Update converter.js

add code to allow no targets

DeepInTheCode committed:
Update index.html

DeepInTheCode committed:
Update index.html

formatting

DeepInTheCode committed:
Update index.html

formatting

DeepInTheCode committed:
Update index.html

formatting

DeepInTheCode committed:
Update DataGridRenderer.js

turn off debugging

DeepInTheCode committed:
Update addTextInputs.js

add blank option

DeepInTheCode committed:
Update converter.js

refactoring

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update addTextInputs.js

added value to options

DeepInTheCode committed:
Update addTextInputs.js

adding

DeepInTheCode committed:
Update addTextInputs.js

check if ddls selected

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

adding mapping

DeepInTheCode committed:
Update converter.js

adding mapping

DeepInTheCode committed:
Update converter.js

debugging

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update converter.js

DeepInTheCode committed:
Update addTextInputs.js

DeepInTheCode committed:
Update addTextInputs.js

changing widths of textboxes and select statements

DeepInTheCode committed:
Update addTextInputs.js

changing widths again

DeepInTheCode committed:
Update addTextInputs.js

testing widths at 90%

DeepInTheCode committed:
Update addTextInputs.js

changing "no mapping" to "none" and widths

abuisman commented:
It would be really helpful if next time you'd write proper commits (messages & structure). Although I might assume that during a Hackathon this isn't really an issue. It does make the code kind of hard to review.

DeepInTheCode commented:
I apologize about that - I was lax in my commenting and commit messages for that very reason.  

Sent from my iPhone

> On Jun 4, 2014, at 2:26, Achilleas notifications@github.com wrote:
> 
> It would be really helpful if next time you'd write proper commits (messages & structure). Although I might assume that during a Hackathon this isn't really an issue. It does make the code kind of hard to review.
> 
> —
> Reply to this email directly or view it on GitHub.

abuisman commented:
@DeepInTheCode No need to apologise of course :) it is just that it would have made reviewing easier. Is there a live demo somewhere? The project sounds cool.

DeepInTheCode commented:
Yes, at http://deepinthecode.github.io/mr-csv-transformer. 

Sent from my iPhone

> On Jun 4, 2014, at 11:27, Achilleas notifications@github.com wrote:
> 
> @DeepInTheCode No need to apologise of course :) it is just that it would have made reviewing easier. Is there a live demo somewhere? The project sounds cool.
> 
> —
> Reply to this email directly or view it on GitHub.

DeepInTheCode committed:
Added conditional checking if there are no targets

DeepInTheCode committed:
Binding change event on "settings" div

Binding change event on "settings" div rather than just the "InputsWrapper" div

DeepInTheCode committed:
Scope of variable, fix CSV format if no targets

DeepInTheCode committed:
Place quotes around strings containing a comma

DeepInTheCode committed:
Added toString() methods to allow use of indexOf()

DeepInTheCode committed:
Create SaveVarAsFile.js

DeepInTheCode committed:
Delete SaveVarAsFile.js

meant to create in JS folder

DeepInTheCode committed:
Create SaveVarAsFile.js

DeepInTheCode committed:
Adding Save button and code to get output text

DeepInTheCode committed:
Adding hidden iframe, script ref. SaveVarAsFile

DeepInTheCode committed:
Added missing ) on line 222

DeepInTheCode committed:
Added outputDataTypes object to convert method

DeepInTheCode committed:
Get file extension at end of convert method

DeepInTheCode committed:
Disabled Save As button on load

DeepInTheCode committed:
Move Save As button

DeepInTheCode committed:
Fixing click event

DeepInTheCode committed:
Define SaveFrame

DeepInTheCode committed:
Declaring scoped variables for output and file ext

DeepInTheCode committed:
Setting visibility of iframe rather than display

DeepInTheCode committed:
Changing id to name and size of hidden iframe

DeepInTheCode committed:
Changing method of finding iframe

DeepInTheCode committed:
Correcting focus and execcommand syntax

DeepInTheCode committed:
Added setTimeout to see if iFrame was rendered yet

DeepInTheCode committed:
Adding id back to iframe

DeepInTheCode committed:
Checking effect of leaving iframe open for focus

DeepInTheCode committed:
Finding element in iFrame for focus

DeepInTheCode committed:
Checking to see if extension is the problem

DeepInTheCode committed:
eliminating focus

DeepInTheCode committed:
closing after execcommand

DeepInTheCode committed:
Trying completely different technique for saving

DeepInTheCode committed:
Code to check for IE, removing extension param

DeepInTheCode committed:
Removing extension option

DeepInTheCode committed:
Removing extra brace

DeepInTheCode committed:
rename variable s to someVar

DeepInTheCode committed:
Changing output from html to plaintext

DeepInTheCode committed:
Removing unnecessary iframe

DeepInTheCode committed:
Adding mongoDB insert format

DeepInTheCode committed:
Adding mongoDB option

DeepInTheCode committed:
Added to description and HTML5 validator link

DeepInTheCode committed:
Fixing typo for HTML5 standard

DeepInTheCode committed:
Changed indentation for readability

DeepInTheCode committed:
Added valid CSS3 logo

DeepInTheCode committed:
Added image

DeepInTheCode committed:
Added shortcut icon

DeepInTheCode committed:
Added target _blank to a tag

DeepInTheCode committed:
Eliminated carriage returns

DeepInTheCode committed:
Clarified instructions on querystring use

DeepInTheCode committed:
Adding vertical-align on header images

DeepInTheCode committed:
Adding text-align to header images

DeepInTheCode committed:
Adding auto height to InputWrapper class

DeepInTheCode committed:
Adding window resize to make div scrollable

DeepInTheCode committed:
Moved resize event

DeepInTheCode committed:
Commented out resize event

DeepInTheCode committed:
Do not include empty columns in mongoDB document

DeepInTheCode committed:
Account for null and empty quotes in rowOutput

DeepInTheCode committed:
Change not-equal/and to equal/or

DeepInTheCode committed:
Eliminate trailing commas from mongoDB jsons

DeepInTheCode committed:
Create enableTab.js

DeepInTheCode committed:
Adding enableTab script to allow tab key in input

DeepInTheCode committed:
Load enableTab.js before converter.js

DeepInTheCode committed:
Not calling enableTab in this file

DeepInTheCode committed:
Enabling tab key in dataInput textarea
